### PR TITLE
#19255 cache region for graphql schema

### PR DESCRIPTION
**Pull request by dsilvam**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/graphql/business/GraphQLSchemaCache.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        cache.put(SCHEMA_KEY, schema, GRAPHQL_SCHEMA_GROUP);
+    }
+
+    public void removeSchema() {
+        cache.remove(SCHEMA_KEY, GRAPHQL_SCHEMA_GROUP);
+    }
+
+    @Override
+    public String[] getGroups() {
+        return GROUPS;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/business/GraphQLSchemaCache.java:43]: Codacy found an issue: [Returning 'GROUPS' may expose an internal array.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6160696)

File: dotCMS/src/main/java/com/dotcms/graphql/business/GraphQLSchemaCache.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    final private DotCacheAdministrator cache = CacheLocator.getCacheAdministrator();
+
+    @Override
+    public String getPrimaryGroup() {
+        return GRAPHQL_SCHEMA_GROUP;
+    }
+
+    public Optional<GraphQLSchema> getSchema() {
+        try {
+            return Optional.ofNullable((GraphQLSchema) cache.get(SCHEMA_KEY, GRAPHQL_SCHEMA_GROUP));
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/business/GraphQLSchemaCache.java:27]: Does `cache.get()` throw a `DotCacheException`?

File: dotCMS/src/main/java/com/dotcms/graphql/business/GraphQLSchemaCache.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.dotmarketing.business.Cachable;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.business.DotCacheException;
+import graphql.schema.GraphQLSchema;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class GraphQLSchemaCache implements Cachable {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/graphql/business/GraphQLSchemaCache.java:11]: doc

File: dotCMS/src/main/java/com/dotcms/graphql/DotGraphQLSchemaProvider.java [hunk 4]
```diff
@@ -23,6 +23,7 @@ public GraphQLSchema getSchema(HandshakeRequest request) {
     @Override
     public GraphQLSchema getSchema() {
         try {
+            APILocator.getGraphqlAPI().printSchema();
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/graphql/DotGraphQLSchemaProvider.java:26]: I think we are missing a test here

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/graphql/business/GraphQLSchemaCache.java (hunk 3 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/business/GraphQLSchemaCache.java:15]: done

File: dotCMS/src/main/java/com/dotcms/graphql/business/GraphQLSchemaCache.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        cache.put(SCHEMA_KEY, schema, GRAPHQL_SCHEMA_GROUP);
+    }
+
+    public void removeSchema() {
+        cache.remove(SCHEMA_KEY, GRAPHQL_SCHEMA_GROUP);
+    }
+
+    @Override
+    public String[] getGroups() {
+        return GROUPS;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/business/GraphQLSchemaCache.java:47]: Codacy found an issue: [Returning 'GROUPS' may expose an internal array.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6160696)

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Review by freddyucv — APPROVED** [review]

**Pull request merged**